packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -412,8 +521,341 @@ describe('plugin-devices', () => {
+            assert.isFalse(devices.registered);
+
+            return devices.refresh()
+              .then(() => assert.called(devices.register));
+          });
+        });
+
+        describe('when the device is registered', () => {
+          let exampleResponse;
+
+          beforeEach('register the device', () => {
+            exampleResponse = {
+              body: {
+                exampleKey: 'example response value'
+              }
+            };
+
+            return devices.register()
+              .then(() => {
+                devices.request = sinon.stub().returns(Promise.resolve(
+                  {...exampleResponse}
+                ));
+              });
+          });
+
+          describe('when the device is ephemeral', () => {
+            beforeEach('set device to ephemeral', () => {
+              devices.config.ephemeral = true;
+            });
+
+            it('should set the ttl property to the config values',
+              () => devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.hasNested(
+                    'body.ttl', devices.config.ephemeralDeviceTTL
+                  ))));
+          });
+
+          describe('when the refresh request is sent', () => {
+            let customHeaders;
+            let customBody;
+
+            beforeEach('configure devices plugin', () => {
+              customHeaders = {
+                testHeader: 'example header value'
+              };
+
+              customBody = {
+                testBody: 'example body value'
+              };
+            });
+
+            it('should allow for custom header key:values', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.headers = {...customHeaders};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.hasNested(
+                    'headers.testHeader', customHeaders.testHeader
+                  )));
+            });
+
+            it('should allow for custom body key:values', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.body = {...customBody};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.hasNested(
+                    'body.testBody', customBody.testBody
+                  )));
+            });
+
+            it('should use the device\'s url property', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.body = {...customBody};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.has(
+                    'uri', devices.url
+                  )));
+            });
+
+            it('should send a PUT request', () => {
+              assert.isTrue(devices.registered);
```

Comment:
These tests from within this describe block are those "setup assertions" that are confusing to read.

Reply:
Moved these like the before comment.